explorer/main.cpp

```diff
@@ -49,6 +50,27 @@ auto ExplorerMain(int argc, char** argv, void* static_for_main_addr,
       "trace_file",
       cl::desc("Output file for tracing; set to `-` to output to stdout."));
 
+  cl::list<ProgramPhase> allowed_program_phases(
+      cl::desc("Select the phases to be added to the output, by default only "
+               "execution trace will be added to the output"),
+      cl::values(
+          clEnumValN(ProgramPhase::SourceProgram, "p_source_program",
+                     "Source Program"),
+          clEnumValN(ProgramPhase::NameResolution, "p_name_resolution",
+                     "Name Resolution"),
+          clEnumValN(ProgramPhase::ControlFlowResolution,
+                     "p_control_flow_resolution", "Control Flow Resolution"),
+          clEnumValN(ProgramPhase::TypeChecking, "p_type_checking",
+                     "Type Checking"),
+          clEnumValN(ProgramPhase::UnformedVariableResolution,
+                     "p_unformed_variables_resolution",
+                     "Unformed Variable Resolutiusing on"),
+          clEnumValN(ProgramPhase::Declarations, "p_declarations",
+                     "Print Declarations"),
+          clEnumValN(ProgramPhase::Execution, "p_execution",
+                     "Program Execution"),
+          clEnumValN(ProgramPhase::Timing, "p_timing", "Timing")));
```

Comment:
I think instead of `"p_<phase>"`, it might be better to call these `"trace_<phase>"`.

Reply:
Done, I also added another flag `-trace_all`, using which users can get the whole output.